react_juce/core/EcmascriptEngine_Hermes.cpp

```diff
@@ -0,0 +1,522 @@
+        Pimpl()
+        {
+            reset();
+        }
+
+        ~Pimpl() = default;
+
+        //==============================================================================
+        juce::var evaluateInline(const juce::String &code)
+        {
+            try
+            {
+                auto jsiBuffer = std::make_shared<jsi::StringBuffer>(code.toStdString());
+                auto js        = runtime->prepareJavaScript(jsiBuffer, "");
+                auto result    = runtime->evaluatePreparedJavaScript(js);
+
+                return jsiValueToVar(result, *runtime);
+            }
+            catch (const jsi::JSIException &e)
+            {
+                throw Error(e.what());
+            }
+        }
+
+        juce::var evaluate(const juce::File &code)
+        {
+            try
+            {
+                auto jsiBuffer = std::make_shared<jsi::StringBuffer>(code.loadFileAsString().toStdString());
+                auto js        = runtime->prepareJavaScript(jsiBuffer, code.getFullPathName().toStdString());
+                auto result    = runtime->evaluatePreparedJavaScript(js);
+
+                return jsiValueToVar(result, *runtime);
+            }
+            catch (const jsi::JSIException &e)
+            {
+                throw Error(e.what());
+            }
+        }
+
+        juce::var evaluateBytecode(const juce::File &code)
+        {
+            try
+            {
+                auto memBlock = std::make_unique<juce::MemoryBlock>();
+                code.loadFileAsData(*memBlock);
+
+                auto jsiBuffer = std::make_shared<JSIMemoryBuffer>(std::move(memBlock));
+                auto js        = runtime->prepareJavaScript(jsiBuffer, code.getFullPathName().toStdString());
+                auto result    = runtime->evaluatePreparedJavaScript(js);
+
+                return jsiValueToVar(result, *runtime);
+            }
+            catch (const jsi::JSIException &e)
+            {
+                throw Error(e.what());
+            }
+        }
+
+        //==============================================================================
+        void registerNativeProperty(const juce::String &name, const juce::var &value)
+        {
+            runtime->global().setProperty(*runtime, name.toRawUTF8(), varToJSIValue(value, *runtime));
+        }
+
+        void registerNativeProperty(const juce::String &target, const juce::String &name, const juce::var &value)
+        {
+            auto obj = runtime->global().getPropertyAsObject(*runtime, target.toRawUTF8());
+            obj.setProperty(*runtime, name.toRawUTF8(), varToJSIValue(value, *runtime));
+        }
+
+        //==============================================================================
+        //TODO: We can probably improve on this by refining the invoke API
+        juce::var invoke(const juce::String &name, const std::vector<juce::var> &vargs)
```

Comment:
@nick-thompson, I think this `invoke` approach is fine for the moment using dot separated accessor strings but it feels a bit cumbersome. Perhaps not essential for this v1 merge but I wonder if we can do something more sophisticated with the `invoke` API down the line. I'm not exactly sure what that would be though ...

Reply:
Agreed. I was thinking initially that the implementation here would basically just call `evaluateInliine` with the string argument, receive a function object back, and then invoke it with the args. But, that's a little cumbersome. 

>  I'm not exactly sure what that would be though ...

Agreed :) Let's keep it as is for now and iterate on this as we have a more clear idea of what the API should be